uk_election_ids/election_ids.py

```diff
@@ -409,3 +367,57 @@ def __repr__(self):
 
     def __eq__(self, other):
         return type(other) == IdBuilder and self.__dict__ == other.__dict__
+
+    @classmethod
+    def parse_from_id(cls, election_id: str) -> "IdBuilder":
+        """
+        Parses a string in to an IdBuilder object
+
+        """
+        if not isinstance(election_id, str):
+            raise ValueError("`identifier` must be a string")
```

Comment:
Three little things:
- Dunno how bothered you are about this, but shouldn't it be a `TypeError`?  
- I think the error message would make more sense to me if it named the parameter that caused the issue. i.e s/identifier/election_id/
- I'm not sure if backticks might have unexpected consequences in rendering...

Reply:
Thanks! `identifier` was from an old name for that arg, so just something I missed in the refactor. 

Do you think catching all three exceptions in the `validate` function is ok?

Reply:
Yeah. `parse_from_id` is a new method, so you're not breaking anything by starting to raise there. Being able to validage and get the errors could also be useful.
Also the `validate` method already caught errors. AFAICT [this](https://github.com/DemocracyClub/EveryElection/blob/65b88befea5f7c3d66a195f5fd82efff050913f6/every_election/apps/api/views.py#L107) is the only place it gets used. I guess it could be useful to return the error, so it can bubble up through to the API response. But also probably fine as it is, and no need to do it here. 

Also just noticed a typo in the commit message 'IdBuidler form'  since you're going to do a fixup anyways :upside_down_face: .

Reply:
(fixed up)